src/command_type.h

```diff
@@ -42,6 +43,16 @@ class CommandCost {
 	 */
 	explicit CommandCost(StringID msg) : expense_type(INVALID_EXPENSES), cost(0), message(msg), success(false), textref_stack_grffile(nullptr), textref_stack_size(0) {}
 
+	/**
+	 * Creates a command return value with two lines of detailed error messages.
+	 */
+	static CommandCost DualErrorMessage(StringID msg, StringID extra_msg)
```

Comment:
I wonder if instead of this static function + initialising extra_message in the class declaration, it'd be better to just use a constructor.

you could even just use the existing constructor with a default INVALID_STRING_ID parameter